app/controllers/admin/internal_docs_controller.rb

```diff
@@ -0,0 +1,43 @@
+class Admin::InternalDocsController < Admin::BaseController
+  before_action :knowledgebase_enabled?, only: ['show']
+  before_action :doc_available_to_view?, only: ['show']
+
+  respond_to :html
+
+  def show
+    define_topic_for_doc
+    define_forum_for_docs
+    generate_page_breadcumbs
+  end
+
+  private
+
+  def doc_available_to_view?
+    @doc = Doc.find_by(id: params[:id], active: true)
+    redirect_to controller: 'errors', action: 'not_found' if @doc.nil? || !@doc.category.internally_viewable?
+  end
+
+  def generate_page_breadcumbs
+    add_breadcrumb t(:knowledgebase, default: "Knowledgebase"), categories_path
+    add_breadcrumb @doc.category.name, category_path(@doc.category) if @doc.category.name
+    add_breadcrumb @doc.title
+    @page_title = @doc.title
+  end
```

Comment:
This is a good way to encapsulate this logic!  Although I didn't see the breadcrumbs used in the view?  DId I miss it?  Also, can this be shared in the non-internal `docs_controller`?